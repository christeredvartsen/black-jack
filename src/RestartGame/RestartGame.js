import React from 'react';
import PropTypes from 'prop-types';

const RestartGame = ({ reset }) => (
  <button onClick={reset}>Restart game</button>
);
RestartGame.propTypes = {
  reset: PropTypes.func.isRequired,
};

export default RestartGame;