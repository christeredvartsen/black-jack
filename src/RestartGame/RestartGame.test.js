import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import RestartGame from './RestartGame';

describe('<RestartGame />', () => {
  test('can reset game', () => {
    const reset = jest.fn();
    shallow(
      <RestartGame reset={reset} />
    ).find('button').simulate('click');
    expect(reset).toHaveBeenCalledTimes(1);
  });

  test('matches snapshot', () => {
    const tree = shallow(
      <RestartGame reset={jest.fn()} />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});