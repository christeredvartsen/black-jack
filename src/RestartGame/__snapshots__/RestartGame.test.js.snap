// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RestartGame /> matches snapshot 1`] = `
<button
  onClick={[MockFunction]}
>
  Restart game
</button>
`;
